core/src/main/java/jenkins/console/ConsoleUrlProviderGlobalConfiguration.java

```diff
@@ -0,0 +1,110 @@
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.Descriptor;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import jenkins.appearance.AppearanceCategory;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.GlobalConfigurationCategory;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Allows administrators to activate and sort {@link ConsoleUrlProvider} extensions to set defaults for all users.
+ * @see ConsoleUrlProviderUserProperty
+ * @since TODO
+ */
+@Extension
+@Symbol("consoleUrlProvider")
+@Restricted(NoExternalUse.class)
+public class ConsoleUrlProviderGlobalConfiguration extends GlobalConfiguration {
+    private static final Logger LOGGER = Logger.getLogger(ConsoleUrlProviderGlobalConfiguration.class.getName());
+
+    private List<ConsoleUrlProvider> providers;
+
+    public ConsoleUrlProviderGlobalConfiguration() {
+        load();
+    }
+
+    @NonNull
+    @Override
+    public GlobalConfigurationCategory getCategory() {
+        return GlobalConfigurationCategory.get(AppearanceCategory.class);
+    }
+
+    public List<ConsoleUrlProvider> getProviders() {
+        return providers;
+    }
+
+    @DataBoundSetter
+    public void setProviders(List<ConsoleUrlProvider> providers) {
+        this.providers = providers;
+        save();
+    }
+
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+        // We have to null out providers before data binding to allow all providers to be deleted in the config UI.
+        // We use a BulkChange to avoid double saves in other cases.
+        try (BulkChange bc = new BulkChange(this)) {
+            providers = null;
+            req.bindJSON(this, json);
+            bc.commit();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Failed to save " + getConfigFile(), e);
+        }
+        return true;
+    }
+
+    public boolean isEnabled() {
+        return ConsoleUrlProvider.isEnabled();
+    }
+
+    public static ConsoleUrlProviderGlobalConfiguration get() {
+        return ExtensionList.lookupSingleton(ConsoleUrlProviderGlobalConfiguration.class);
+    }
+
+    public List<? extends Descriptor<ConsoleUrlProvider>> getProvidersDescriptors() {
+        // For the global configuration, the default provider will always be consulted as a last resort, and since it
+        // handles all builds, there is no reason to ever select it explicitly.
+        return Jenkins.get().getDescriptorList(ConsoleUrlProvider.class).stream()
+                .filter(d -> !(d instanceof DefaultConsoleUrlProvider.DescriptorImpl))
```

Comment:
it seems odd that this is filtered out here but not on the user property, adding it doesn't seem to make any difference on the user property, it will still always fall back to it

<img width="281" alt="image" src="https://github.com/jenkinsci/jenkins/assets/21194782/ce3b594b-5375-4047-9663-1b135a6e4e49">

Reply:
In the global view, there's no difference between configuring none, and configuring Default. IMO it's confusing to have two different configurations that behave identically, so it makes sense to not be visible there.

In the user view, it's there to allow users to explicitly select Default no matter what the admin configures. It needs to exist because there's no other way for users to accomplish this.

Reply:
Ah I see that makes sense.